src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,201 @@
+     */
+    public static Command parse(String fullCmd) throws DukeException {
+        String cmd = fullCmd.split(" ")[0];
+        switch (cmd) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+            return parseMark(fullCmd, true);
+        case "unmark":
+            return parseMark(fullCmd, false);
+        case "todo":
+            return parseToDo(fullCmd);
+        case "deadline":
+            return parseDeadline(fullCmd);
+        case "event":
+            return parseEvent(fullCmd);
+        case "delete":
+            return parseDelete(fullCmd);
+        case "find":
+            return parseFind(fullCmd);
+        default:
+            throw new DukeException("\tSorry, what do you mean?\n");
+        }
+    }
+
+    private static MarkCommand parseMark(String fullCmd, boolean isComplete)
+            throws DukeException {
+        String markFormTxt = "\tSorry! To mark or unmark tasks, please do\n"
+                + "\t\t(un)mark (number)\n";
+        if (fullCmd.split(" ")[1].trim().isEmpty()) {
+            throw new DukeException(markFormTxt);
+        }
+        int updateIndex = Integer.parseInt(fullCmd.split(" ")[1]);
+        return new MarkCommand(updateIndex, isComplete);
+    }
+
+    private static AddCommand parseDeadline(String fullCmd) throws DukeException {
+        String deadlineFormTxt = "\tSorry! Please use the given format for deadline tasks:\n"
+                + "\t\tdeadline (description) /by DD/MM/YYYY hhmm\n"
+                + "\t\teg. deadline homework /by 30/01/2023 1800\n";
+
+        try {
+            String[] fullCmdArr = fullCmd.split("/", 2);
+            if (fullCmdArr.length != 2) {
+                throw new DukeException(deadlineFormTxt);
+            }
+            if (!fullCmdArr[1].substring(0, Math.min(3, fullCmdArr[1].length())).equals("by ")) {
+                throw new DukeException(deadlineFormTxt);
+            }
+
+            String name = fullCmdArr[0].substring(9);
+            String dueDate = fullCmd.split("/", 2)[1].substring(3);
+            if (name.trim().isEmpty()) {
+                throw new DukeException("Description Blank");
+            }
+            if (dueDate.trim().isEmpty()) {
+                throw new DukeException(deadlineFormTxt);
+            }
+
+            Task newTask = new Deadline(name, parseCmdDate(dueDate));
+            return new AddCommand(newTask);
+        } catch (Exception e) {
+            throw new DukeException(deadlineFormTxt);
+        }
+    }
+
+    private static AddCommand parseEvent(String fullCmd) throws DukeException {
+        String eventFormTxt = "\tSorry! Please use the given format for event tasks:\n"
+                + "\t\tevent (description) /from DD/MM/YYYY hhmm /to DD/MM/YYYY hhmm\n"
+                + "\t\teg. event assignment /from 30/01/2023 1200 /to 12/02/2023 2359\n";
```

Comment:
ooh that's very interesting way of storing the format of the string for replacement later!